chain/client/src/client.rs

```diff
@@ -2004,4 +2012,63 @@ impl Client {
             next_blocks_log.join("\n")
         ))
     }
+
+    pub fn detailed_upcoming_blocks_info_as_web(&self) -> ChunkInfoView {
+        let height_status_map = match self.detailed_upcoming_blocks_info() {
+            Ok(m) => m,
+            Err(_) => HashMap::new(),
+        };
+
+        let next_blocks_log = height_status_map
+            .keys()
+            .sorted()
+            .map(|height| {
+                let val = height_status_map.get(height).unwrap();
+
+                let block_debug = val
+                    .iter()
+                    .map(|entry| {
+                        let block_info = entry.1;
+                        let chunk_status = block_info
+                            .chunk_hashes
+                            .iter()
+                            .map(|it| {
+                                if block_info.chunks_completed.contains(it) {
+                                    "(done)"
```

Comment:
nit: if you're doing it as a string -- you might want to make sure that all the values here have the same length.

otherwise it will look bad on the webpage.

do you have a demo somewhere of how it looks?

Reply:
Cool. Let me change this to be all using four characters and try using fixed fonts.